storage/innobase/ddl/ddl0builder.cc

```diff
@@ -2047,7 +2047,7 @@ dberr_t Builder::finish() noexcept {
   }
 
   dberr_t err{DB_SUCCESS};
-  if (get_error() == DB_SUCCESS) {
+  if (get_error() == DB_SUCCESS && !m_index->table->is_temporary()) {
```

Comment:
Don't we do bulk load for ADD INDEX on temporary tables?
isn't there online log to apply for temporary tables?

Reply:
we do bulk load. IIUC, there wont' be online log to apply on temporary tables since these tables are per session.

Reply:
IMHO, we should only skip writing redo. Flushing should be done

Reply:
for temp tables

Reply:
Flushing would ensure data is persistant to disk. but in the case of temporary table we don't need that. if server crashes. we can safely discard that table.

Reply:
Hmm, you are right, nobody copies this datafile and we don't need to necessarily wait. Can you please check without your patch if observer->flush() is called or not for temp tables?

Reply:
no we do not Flush_observer::flush() for temp tables.